### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by degasus**
Two small and trivial 1% optimizations. Please read the commit messages of the patches.

### CODE REVIEW

File: src/video_core/command_processor.cpp [hunk 1]
```diff
@@ -47,8 +49,6 @@ void GPU::WriteReg(u32 method, u32 subchannel, u32 value, u32 remaining_params)
         return;
     }
 
-    ASSERT(bound_engines.find(subchannel) != bound_engines.end());
-
     const EngineID engine = bound_engines[subchannel];
```
Comment (degasus) [inline src/video_core/command_processor.cpp]: It is also possible to store the function pointers in the std:array. But I'm not sure if the switch is slower or faster than this kind of indirect branch.

File: src/core/core.cpp [hunk 2]
```diff
@@ -441,8 +441,8 @@ void System::SetGPUDebugContext(std::shared_ptr<Tegra::DebugContext> context) {
     impl->debug_context = std::move(context);
 }
 
-std::shared_ptr<Tegra::DebugContext> System::GetGPUDebugContext() const {
-    return impl->debug_context;
+Tegra::DebugContext* System::GetGPUDebugContext() const {
```
Comment (lioncash) [inline src/core/core.cpp]: This can be a reference, given it's required to be valid for the entire system execution lifetime.

Comment (degasus) [reply src/core/core.cpp]: Reference to the shared_ptr? AFAIK the object itself might be a nullptr.

Comment (lioncash) [reply src/core/core.cpp]: Ah true, disregard my comment then. I forgot that the command-line version of yuzu doesn't set a debug context.

**Pull request merged**